cocos/audio/apple/AudioEngine-inl.mm

```diff
@@ -710,6 +711,11 @@ AL_API ALvoid AL_APIENTRY alGenBuffers(ALsizei n, ALuint *bids)
 void AudioEngineImpl::uncacheAll()
 {
     _audioCaches.clear();
+    for(auto&& player : _audioPlayers)
+    {
+        // prevent player hold invalid AudioCache* pointer, since all audio caches purged
+        player.second->setCache(nullptr);
```

Comment:
AudioEngine::uncacheAll/uncache will invoke AudioEngineImpl::stopAll/stop first, it means that AudioPlayer will be destroyed  before AudioEngineImpl::uncacheAll is invoked.

Reply:
it just means `player->destroy()` called before uncacheAll is invoked, so `player->_audioCache` is wild pointer before player destructed invoked.

Reply:
at that short time, the `rotateBufferThread ` should been stopped, if not, the invalid cache pointer will case thread crash too.